Using QueuingTransactionReceiptProcessor throws exception when Contract.executeTransaction is called

# Using QueuingTransactionReceiptProcessor throws exception when Contract.executeTransaction is called

## Steps To Reproduce
Use QueuingTransactionReceiptProcessor  and call any contract transactional method.

### Expected behavior
`UnsupportedOperationException` should not be thrown.

### Actual behavior
the following exception is thrown:
```
	java.lang.UnsupportedOperationException: Empty transaction receipt, only transaction hash is available
	at org.web3j.tx.response.EmptyTransactionReceipt.unsupportedOperation(EmptyTransactionReceipt.java:43)
	at org.web3j.tx.response.EmptyTransactionReceipt.getStatus(EmptyTransactionReceipt.java:139)
	at org.web3j.protocol.core.methods.response.TransactionReceipt.isStatusOK(TransactionReceipt.java:161)
	at org.web3j.tx.Contract.executeTransaction(Contract.java:372)
```

## Environment
_Describe the environment in which the issue occurs_
- Web3j version: 4.5.18. 
- Java 11
- Operating System: any


## Additional context
This broke in this commit: https://github.com/web3j/web3j/commit/99e9df1e386a13403a53f2d36009409bc52b4efb#diff-a71774ffb4764fa21a66e2f3dbbe40f7
The code in `TransactionReceipt` was changed from:
```
    public boolean isStatusOK() {
        if (null == status) {
            return true;
        }
```
to 
```
    public boolean isStatusOK() {
        if (null == getStatus()) { // <-- call to getter method that throws the exception
            return true;
        }
```

`EmptyTransactionReceipt` throws `UnsupportedOperationException` on `getStatus()`.

Possible solutions:
Either revert the call to `getStatus()` or override `isStatusOK()` in `EmptyTransactionReceipt`.

Same issue when using NoOpProcessor.

```
CompletableFuture<TransactionReceipt> transactionInProgress = someContract.doSomething(...);
transactionInProgress.thenAccept(...).exceptionally(...);
```
Always leads to the following exception:
```
java.util.concurrent.CompletionException: java.lang.UnsupportedOperationException: Empty transaction receipt, only transaction hash is available
        at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:331)
        at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:346)
        at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:704)
        at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
        at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2158)
        at org.web3j.utils.Async.lambda$run$1(Async.java:40)
        at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1800)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
        at java.base/java.lang.Thread.run(Thread.java:831)
Caused by: java.lang.UnsupportedOperationException: Empty transaction receipt, only transaction hash is available
        at org.web3j.tx.response.EmptyTransactionReceipt.unsupportedOperation(EmptyTransactionReceipt.java:43)
        at org.web3j.tx.response.EmptyTransactionReceipt.getStatus(EmptyTransactionReceipt.java:139)
        at org.web3j.protocol.core.methods.response.TransactionReceipt.isStatusOK(TransactionReceipt.java:161)
        at org.web3j.tx.Contract.executeTransaction(Contract.java:375)
```

Problem is here in Contract.java:375:
```
TransactionReceipt receipt =
                send(
                        contractAddress,
                        data,
                        weiValue,
                        gasProvider.getGasPrice(funcName),
                        gasProvider.getGasLimit(funcName),
                        constructor);

        if (!receipt.isStatusOK()) { // throws if EmptyTransactionReceipt
```

--- a/core/src/main/java/org/web3j/tx/Contract.java
+++ b/core/src/main/java/org/web3j/tx/Contract.java
@@ -49,6 +49,7 @@ import org.web3j.tx.exceptions.ContractCallException;
 import org.web3j.tx.gas.ContractEIP1559GasProvider;
 import org.web3j.tx.gas.ContractGasProvider;
 import org.web3j.tx.gas.StaticGasProvider;
+import org.web3j.tx.response.EmptyTransactionReceipt;
 import org.web3j.utils.Numeric;
 
 import static org.web3j.utils.RevertReasonExtractor.extractRevertReason;
@@ -403,7 +404,9 @@ public abstract class Contract extends ManagedTransaction {
             throw new TransactionException(error.getData().toString());
         }
 
-        if (receipt != null && !receipt.isStatusOK()) {
+        if (!(receipt instanceof EmptyTransactionReceipt)
+                && receipt != null
+                && !receipt.isStatusOK()) {
             throw new TransactionException(
                     String.format(
                             "Transaction %s has failed with status: %s. "

--- a/core/src/test/java/org/web3j/tx/ContractTest.java
+++ b/core/src/test/java/org/web3j/tx/ContractTest.java
@@ -51,6 +51,7 @@ import org.web3j.tx.gas.ContractGasProvider;
 import org.web3j.tx.gas.DefaultGasProvider;
 import org.web3j.tx.gas.StaticEIP1559GasProvider;
 import org.web3j.tx.gas.StaticGasProvider;
+import org.web3j.tx.response.EmptyTransactionReceipt;
 import org.web3j.utils.Async;
 import org.web3j.utils.Numeric;
 
@@ -577,6 +578,19 @@ public class ContractTest extends ManagedTransactionTester {
 
         assertEquals(eventValuesWithLogs2.size(), 1);
         assertEquals(eventValuesWithLogs2.get(0).getLog(), logs.get(1));
+    }
+
+    @Test
+    public void testEmptyTransactionReceipt() throws Exception {
+        TransactionReceipt transactionReceipt = new EmptyTransactionReceipt(TRANSACTION_HASH);
+
+        prepareTransaction(transactionReceipt);
+
+        assertEquals(
+                transactionReceipt,
+                contract.performTransaction(
+                                new Address(BigInteger.TEN), new Uint256(BigInteger.ONE))
+                        .send());
     }
 
     void testErrorScenario() throws Throwable {

